sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/PlanHelper.scala

```diff
@@ -45,6 +45,7 @@ object PlanHelper {
         case e: AggregateExpression
           if !(plan.isInstanceOf[Aggregate] ||
                plan.isInstanceOf[Window] ||
+               plan.isInstanceOf[LateralJoin] ||
```

Comment:
`LateralJoin` can host aggregate expressions?

Reply:
This is because the children (outerAttrs) field can contain aggregate expressions `LateralJoin(left, LateralSubquery(plan, [sum(a)]), ..)` when collecting outer references in the subquery:

https://github.com/apache/spark/blob/dfd8a8dc676c388c0c1bb7e4cb8d55eab10504ab/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala#L201-L211

Reply:
How about more strictly checking that the right plan exprs of the lateral join are aggr ones?